contracts/SnappAuction.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.5.0;
+
+import "./SnappBase.sol";
+
+
+contract SnappAuction is SnappBase {
+  
+    uint16 public constant AUCTION_BATCH_SIZE = 1000;
+
+    uint public auctionIndex;
+    mapping (uint => PendingFlux) public auctions;
+
+    event SellOrder(
+        uint auctionId, 
+        uint16 slotIndex,
+        uint16 accountId, 
+        uint8 buyToken, 
+        uint8 sellToken, 
+        uint128 buyAmount,
+        uint128 sellAmount
+    );
+
+    constructor () public {
+        auctions[auctionIndex].creationBlock = block.number;
+    }
+
+    function placeSellOrder(
+        uint8 buyToken,
+        uint8 sellToken,
+        uint128 minBuyAmount,
+        uint128 maxSellAmount
+    ) public onlyRegistered() {
+        require(buyToken != sellToken, "Buy and Sell tokens must differ!");
+
+        // Must have 0 < tokenId < MAX_TOKENS anyway, so might as well ensure registered.
+        require(tokenIdToAddressMap[buyToken] != address(0), "Buy token is not registered");
+        require(tokenIdToAddressMap[sellToken] != address(0), "Sell token is not registered");
+
+        // Restrict buy and sell amount to occupy less than 100 bits.
+        require(minBuyAmount < 0x10000000000000000000000000, "Buy amount too large!");
+        require(maxSellAmount < 0x10000000000000000000000000, "Sell amount too large!");
+
+        if (
+            auctions[auctionIndex].size == AUCTION_BATCH_SIZE || 
+            block.number > auctions[auctionIndex].creationBlock + 20
+        ) {
+            auctionIndex++;
+            auctions[auctionIndex] = PendingFlux({
+                size: 0,
+                shaHash: bytes32(0),
+                creationBlock: block.number,
+                appliedAccountStateIndex: 0
```

Comment:
Is this relevant for orders? Will it be the state index in which they are processed?

Reply:
Good question. I wonder if this "pendingBatch" should have attributes for `prices` and `fulfilmentVolumes`. Then state transition can be performed based on the batch itself and the `appliedStateIndex` would actually make sense. This, however, would be a change to make in another PR (related to state transition from auction results).

Reply:
If we see this data structure to be different from deposits/orders I'd suggest creating a separate struct and calling it PendingOrderBatch (and leaving PendingFlux). However if the structures will be the same for the foreseeable future, I think we should use the same type.

Reply:
At this moment (for this task), I see them as being the same. However, this could very-well change in the next task (when state transition resulting from auctions is implemented). Fortunately, that is not the purpose of this PR, so it will remain as is until a change is necessary.

It seems as if we may not want to store prices and volumes on the contract anyway, since this could, technically, be recovered from the transaction payload.